src/transaction/log_lsa.hpp

```diff
@@ -29,12 +29,18 @@
 
 #include <cassert>
 #include <cinttypes>
+#include <cstring>
 #include <cstddef>
 
+const int page_id_bit_count = 48;
+const int page_offset_bit_count = 16;
+
 struct log_lsa
 {
-  std::int64_t pageid:48;		/* Log page identifier : 6 bytes length */
-  std::int64_t offset:16;		/* Offset in page : 2 bytes length.
+std::int64_t pageid:
+  page_id_bit_count;		/* Log page identifier : 6 bytes length */
+std::int64_t offset:
+  page_offset_bit_count;		/* Offset in page : 2 bytes length.
```

Comment:
I find previous code more readable. It was easily understandable that pageid+offset take 8 bytes.

Usually it is best to give literals names to help other devs understand the literal meaning. But this is a bit_field and it is expected to see the literals. Please notice that the code in the standard [bit_field](https://en.cppreference.com/w/cpp/language/bit_field) manual does the same.

Later edit: I see now that the constants are used for the cast operators. The same kind of result can be obtained with the help of std::memcpy. In my opinion, std::memcpy version will be friendlier with the future reader.